grammars/git commit message.cson

```diff
@@ -23,26 +23,37 @@
         'name': 'comment.line.number-sign.git-commit'
       }
       {
-        # Subject line should be capitalized, see:
-        # http://chris.beams.io/posts/git-commit/#capitalize
-        'match': '\\A[a-z]'
-        'name': 'invalid.illegal.first-char-should-be-uppercase.git-commit'
+        # Subject line 0-50 chars
+        'match': '\\A(?!#)(([a-z])|.).{0,48}((\\.)|.)$'
+        'captures':
+          '2':
+            'name': 'invalid.illegal.first-char-should-be-uppercase.git-commit'
+          '4':
+            'name': 'invalid.illegal.subject-no-trailing-period.git-commit'
       }
       {
-        # Subject line 70 chars or longer
-        'match': '\\A(?!#).{50}(.{19})(.+)'
+        # Subject line 51-69 chars
+        'match': '\\A(?!#)(([a-z])|.).{49}(.{0,18})((\\.)|(.))$'
         'captures':
-          '1':
-            'name': 'invalid.deprecated.line-too-long.git-commit'
           '2':
-            'name': 'invalid.illegal.line-too-long.git-commit'
+            'name': 'invalid.illegal.first-char-should-be-uppercase.git-commit'
+          '3':
+            'name': 'invalid.deprecated.line-too-long.git-commit'
+          '5':
+            'name': 'invalid.illegal.subject-no-trailing-period.git-commit'
+          '6':
+            'name': 'invalid.deprecated.line-too-long.git-commit'
       }
       {
-        # Subject line 51-69 chars
-        'match': '\\A(?!#).{50}(.{0,19})'
+        # Subject line 70 chars or longer
+        'match': '\\A(?!#)(([a-z])|.).{49}(.{0,19})(.+)$'
```

Comment:
I don't think it can actually be both; my question was really related to what `name:` should be.

Right now the `name:` for a post-70-chars trailing period would be `invalid.illegal.line-too-long.git-commit` but not `invalid.illegal.subject-no-trailing-period.git-commit`.

With your suggested change the `name:` for a post 70-chars-trailing period would be `invalid.illegal.subject-no-trailing-period.git-commit` but not `invalid.illegal.line-too-long.git-commit`.

Or can `name:` take an array? The only documentation I managed to find doesn't seem to think so, but maybe it is possible? Do you have a better documentation link (as in "for Atom, not for TextMate")?
https://manual.macromates.com/en/language_grammars#rule_keys

So if we can't have both, which is more important for a post-70-chars trailiing period, `invalid.illegal.subject-no-trailing-period.git-commit` or `invalid.illegal.line-too-long.git-commit`?